ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/TableExport.java

```diff
@@ -152,7 +152,7 @@ private void writeData(PartitionIterable partitions) throws SemanticException {
       if (tableSpec.tableHandle.isPartitioned()) {
         if (partitions == null) {
           throw new IllegalStateException("partitions cannot be null for partitionTable :"
-              + tableSpec.getTableName().getTable());
+              + tableSpec.getTableName().getNotEmptyDbTable());
```

Comment:
would it make a lot of changes if we would rely on TableName's toString() method for cases like this....I don't think we should retain the old exception messages at any cost....

Reply:
I guess using the logic of `getNotEmptyDbTable` for `toString` would make the most sense then.